controllers/etcd_controller.go

```diff
@@ -735,83 +727,110 @@ func (r *EtcdReconciler) getMapFromEtcd(etcd *druidv1alpha1.Etcd) (map[string]in
 		values["tlsCASecret"] = etcd.Spec.Etcd.TLS.TLSCASecretRef.Name
 	}
 
+	if etcd.Spec.Backup.Store != nil {
+		storeValues := map[string]interface{}{
+			"storageContainer": etcd.Spec.Backup.Store.Container,
+			"storePrefix":      etcd.Spec.Backup.Store.Prefix,
+			"storageProvider":  utils.StorageProviderFromInfraProvider(etcd.Spec.Backup.Store.Provider),
+			"storeSecret":      etcd.Spec.Backup.Store.SecretRef.Name,
+		}
+		values["store"] = storeValues
+	}
+
 	return values, nil
 }
 
 func (r *EtcdReconciler) addFinalizersToDependantSecrets(etcd *druidv1alpha1.Etcd) error {
-
-	storeSecret := corev1.Secret{}
-	r.Client.Get(context.TODO(), types.NamespacedName{
-		Name:      etcd.Spec.Backup.Store.SecretRef.Name,
-		Namespace: etcd.Namespace,
-	}, &storeSecret)
-	if finalizers := sets.NewString(storeSecret.Finalizers...); !finalizers.Has(FinalizerName) {
-		logger.Infof("Adding finalizer (%s) for secret %s by etcd (%s)", FinalizerName, storeSecret.GetName(), etcd.Name)
-		storeSecretCopy := storeSecret.DeepCopy()
-		finalizers.Insert(FinalizerName)
-		storeSecretCopy.Finalizers = finalizers.UnsortedList()
-		if err := r.Update(context.TODO(), storeSecretCopy); err != nil {
+	if etcd.Spec.Backup.Store != nil {
+		storeSecret := corev1.Secret{}
+		err := r.Client.Get(context.TODO(), types.NamespacedName{
+			Name:      etcd.Spec.Backup.Store.SecretRef.Name,
+			Namespace: etcd.Namespace,
+		}, &storeSecret)
+		if err != nil && !errors.IsNotFound(err) {
 			return err
 		}
+		if err == nil {
```

Comment:
What if the secret is not found?

Reply:
It would fail at statefulset creation as pod will not be able to mount the secret. Shall I error out here?